pkg/cmd/provisioning/aws/create_identity_provider.go

```diff
@@ -279,22 +282,69 @@ func createIdentityProvider(client aws.Client, name, region, publicKeyPath, targ
 	return identityProviderARN, nil
 }
 
+func getTLSCertificatesWithProxy(urlWithPort, proxyEnv string) ([]*x509.Certificate, error) {
+
+	proxyUrl, err := url.Parse(proxyEnv)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse HTTP_PROXY URL %s", urlWithPort)
+	}
+
+	// Create transport to setup proxy later.
+	tlsConfig := http.DefaultTransport.(*http.Transport).TLSClientConfig
+	transport := &http.Transport{
+		DialTLS: func(network, addr string) (net.Conn, error) {
+			conn, err := tls.Dial(network, addr, tlsConfig)
+			return conn, err
+		},
+	}
+	log.Printf("Using HTTP_PROXY=%s", proxyUrl)
+	transport.Proxy = http.ProxyURL(proxyUrl)
+	urlWithPort = fmt.Sprintf("https://%s", urlWithPort)
+
+	c := &http.Client{Transport: transport}
+	resp, err := c.Get(urlWithPort)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to validate URL %s to check existing Identity Provider", urlWithPort)
+	}
+	if resp.TLS == nil {
+		return nil, errors.Wrapf(err, "unable to get TLS connection from URL %s", urlWithPort)
+	}
+	if resp.TLS.PeerCertificates == nil {
+		return nil, errors.Wrapf(err, "unable to get TLS PeerCertificates from connection URL %s", urlWithPort)
+	}
+
+	return resp.TLS.PeerCertificates, nil
+}
+
+func getTLSCertificates(urlWithPort string) ([]*x509.Certificate, error) {
+	conn, err := tls.Dial("tcp", urlWithPort, &tls.Config{})
+	if err != nil {
+		return nil, err
+	}
+
+	return conn.ConnectionState().PeerCertificates, nil
+}
+
 func getTLSFingerprint(bucketURL string) (string, error) {
 	u, err := url.Parse(bucketURL)
 	if err != nil {
 		return "", err
 	}
 
+	var certs []*x509.Certificate
 	urlWithPort := fmt.Sprintf("%s:443", u.Host)
 
-	conn, err := tls.Dial("tcp", urlWithPort, &tls.Config{})
+	if proxyEnv := os.Getenv("HTTP_PROXY"); proxyEnv != "" {
+		log.Printf("Found proxy environment variable HTTP_PROXY=%s", proxyEnv)
+		certs, err = getTLSCertificatesWithProxy(urlWithPort, proxyEnv)
+	} else {
+		certs, err = getTLSCertificates(urlWithPort)
+	}
```

Comment:
I believe all of this can be simplified by a) converging both paths to use `resp, err := c.Get(urlWithPort)`, and b) using the default client `client := http.DefaultClient`. My understanding is that [DefaultClient](https://pkg.go.dev/net/http#Client) uses [DefaultTransport](https://pkg.go.dev/net/http#RoundTripper), which in turn uses [ProxyFromEnvironment](https://pkg.go.dev/net/http#ProxyFromEnvironment). This should pull in the environment variables HTTP_PROXY, HTTPS_PROXY, and NO_PROXY in a consistent way.

```diff
-	if proxyEnv := os.Getenv("HTTP_PROXY"); proxyEnv != "" {
-		log.Printf("Found proxy environment variable HTTP_PROXY=%s", proxyEnv)
-		certs, err = getTLSCertificatesWithProxy(urlWithPort, proxyEnv)
-	} else {
-		certs, err = getTLSCertificates(urlWithPort)
-	}
- 	if err != nil {
-		return "", errors.Wrapf(err, "error validating TLS Fingerprint")
-	}
+	client := http.DefaultClient
+	resp, err := client.Get(urlWithPort)
+	if err != nil {
+		return "", errors.Wrapf(err, "error validating TLS Fingerprint")
+	}
+	defer resp.Body.Close()
+	
+	if resp.TLS == nil {
+		return nil, errors.Wrapf(err, "unable to get TLS connection from URL %s", urlWithPort)
+	}
+	if resp.TLS.PeerCertificates == nil {
+		return nil, errors.Wrapf(err, "unable to get TLS PeerCertificates from connection URL %s", urlWithPort)
+	}
+
+	certs := resp.TLS.PeerCertificates
```

Reply:
Thanks, feedback provided. Tested successfully in private environment with proxy vars set.